### PR TITLE
fix(VOverlay): account for max-width from props

### LINKED ISSUE #14420: [Bug Report][3.0.0-alpha.11] Prop `max-width` of VTooltip component has no effect
### Environment
**Vuetify Version:** 3.0.0-alpha.11
**Vue Version:** 3.2.21
**Browsers:** Chrome 96.0.4664.45
**OS:** Windows 10

### Steps to reproduce
See repro. Hover over the button.

### Expected Behavior
The tooltip width should be limited to 300px.

### Actual Behavior
It's not.

### Reproduction Link
<a href="https://codepen.io/glen-84/pen/NWgvKOw" target="_blank">https://codepen.io/glen-84/pen/NWgvKOw</a>

### Other comments
From Kael on Discord:

> looks like a bug, there is a max-width prop but it's overridden by the computed width available on that side

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #14420

## Description

fixes #14420

## How Has This Been Tested?

visually | unit 
## Markup:

<details>

```vue
// Paste your FULL Playground.vue here
<template>
  <v-container>
    <v-btn>
        <v-tooltip
            :text="longTooltip"
            activator="parent"
            max-width="300" />
        Example
    </v-btn>
  </v-container>
</template>

<script>
export default {
  data: () => ({
    longTooltip: "x".repeat(200)
  }),
};
</script>


```
</details>

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Improvement/refactoring (non-breaking change that doesn't add any features but makes things better)

## Checklist:

- [x] The PR title is no longer than 64 characters.
- [x] The PR is submitted to the correct branch (`master` for bug fixes and documentation updates, `dev` for new features and backwards compatible changes and `next` for non-backwards compatible changes).
- [x] My code follows the code style of this project.
- [ ] I've added relevant changes to the documentation (applies to new features and breaking changes in core library)

### CODE REVIEW

**Comment by KaelWD** [comment]
This probably needs to be changed in the position strategy instead - if there's only 250px of space available but I've set `max-width="300"`, what should the width be?

**Comment by yuwu9145** [comment]
Makes a lot of sense. Changes are switched to using position strategy

File: packages/vuetify/src/components/VOverlay/positionStrategies.ts [hunk 1]
```diff
@@ -208,10 +208,15 @@ function connectedPositionStrategy (data: PositionStrategyData, props: StrategyP
 
     const canFill = doesOverlap.value || ['center', 'top', 'bottom'].includes(anchor.side)
 
+    const maxWidthFromProps: number = !props.maxWidth ? Number.MAX_SAFE_INTEGER // use this constant to indicate no max width limit
```
Comment (johnleider) [inline packages/vuetify/src/components/VOverlay/positionStrategies.ts:211]: Is it necessary to use this? I don't see a specific benefit to it.

Comment (yuwu9145) [reply packages/vuetify/src/components/VOverlay/positionStrategies.ts:211]: Without this, when `maxWidth` is not set in props, `maxWidthFromProps` will be `undefined` which can cause `Math.min(undefined, 1)` to return `NaN` in line 216 & 217.

Comment (yuwu9145) [reply packages/vuetify/src/components/VOverlay/positionStrategies.ts:211]: Probably better to allow `undefined` here and do Falsy conditional check in 216 & 217.

Leave with me and I will optimise

**Comment by yuwu9145** [comment]
@johnleider @KaelWD is there any objection to upgrade `@vue/test-utils` to version `rc.17` ? 

It will make writing tests easier with components using `teleport` like `VTooltip` & `VOverlay`. Here is the [detail](https://github.com/vuejs/test-utils/issues/1266)

**Comment by yuwu9145** [comment]
could I please ask for a feedback for this PR? @KaelWD

**Review by johnleider — CHANGES_REQUESTED** [review]
I think this implementation is more complex than it needs to be. Only needs to be 3 new lines:

image

Also, please remove the VTooltip test changes. Thank you!

**Comment by yuwu9145** [comment]
> I think this implementation is more complex than it needs to be. The following accomplishes the same thing.
> 
> image
> 
> Also, please remove the VTooltip test changes. Thank you!

Hi @johnleider , thanks for your tip. But it does not cover the senario when `props.maxWidth` is `undefined`.

I think we should regard undefined maxWidth as maximally occupying whole remaining space by default. Pull request has been updated to achieve this on top of your changes.

Thank you!

**Comment by johnleider** [comment]
> Hi @johnleider , thanks for your tip. But it does not cover the senario when props.maxWidth is undefined.

Then add that one use-case. The general point I was trying to make was that I did not like your implementation and wanted you to approach it differently.

**Comment by yuwu9145** [comment]
> > Hi @johnleider , thanks for your tip. But it does not cover the senario when props.maxWidth is undefined.
> 
> 
> 
> Then add that one use-case. The general point I was trying to make was that I did not like your implementation and wanted you to approach it differently.

No worries, understand. Pull request has been updated based on your advice. Feel free to leave any further comments

File: packages/vuetify/src/components/VOverlay/positionStrategies.ts [hunk 2]
```diff
@@ -212,6 +215,7 @@ function connectedPositionStrategy (data: PositionStrategyData, props: StrategyP
       : anchor.side === 'end' ? freeSpace.right
       : anchor.side === 'start' ? freeSpace.left
       : null
+
```
Comment (johnleider) [inline packages/vuetify/src/components/VOverlay/positionStrategies.ts:218]:
```suggestion
```

**Review by johnleider — CHANGES_REQUESTED** [review]

**Review by johnleider — APPROVED** [review]
Looks good. Just need review from @KaelWD

**Pull request merged**